google/cloud/storage/internal/make_options_span.h

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_MAKE_OPTIONS_SPAN_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_MAKE_OPTIONS_SPAN_H
+
+#include "google/cloud/storage/version.h"
+#include "google/cloud/options.h"
+#include <utility>
+
+namespace google {
+namespace cloud {
+namespace storage {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace internal {
+
+///@{
+/**
+ * The `GroupOptions()` overload set groups all the `google::cloud::Options`
+ * present in a parameter pack into a single `google::cloud::Options`.
```

Comment:
Perhaps mention that option values later in the pack are preferred when there are duplicates?

Reply:
Done.